File: src/main/java/tk/skulk/plugin/core/extensions/silencemobs/SilenceMobsExtension.java
package tk.skulk.plugin.core.extensions.silencemobs;

import org.jetbrains.annotations.NotNull;
import tk.skulk.plugin.core.Plugin;
import tk.skulk.plugin.core.extensions.silencemobs.listeners.PlayerInteractEntityListener;
import tk.skulk.plugin.wrapper.BaseExtension;

public final class SilenceMobsExtension extends BaseExtension {
    public SilenceMobsExtension(final @NotNull Plugin plugin) {
        super(plugin);
    }

    @Override protected void initListeners() {
        new PlayerInteractEntityListener(this);
    }
}

File: src/main/kotlin/tk/skulk/plugin/core/extensions/silenceMobs/SilenceMobsExtension.kt
package tk.skulk.plugin.core.extensions.silenceMobs

import tk.skulk.plugin.core.SNPlugin
import tk.skulk.plugin.core.extensions.silenceMobs.listeners.PlayerInteractEntityListener
import tk.skulk.plugin.framework.Extension

class SilenceMobsExtension(plugin: SNPlugin) : Extension<SNPlugin>(plugin) {
    override val listeners = listOf {
        PlayerInteractEntityListener(this)
    }
}
